docs/2-building-blocks-of-oop-part-1/hometask/src/Point.ts

```diff
@@ -1 +1,39 @@
-export class Point {}
+export class Point {
+  private x: number;
+  private y: number;
+
+  private calculateDistanceTo(x: number, y: number): number {
+    const distance: number = Math.sqrt(
+      Math.pow(this.x - x, 2) + Math.pow(this.y - y, 2)
+    );
+
+    return parseFloat(distance.toFixed(2));
+  }
+
+  constructor();
+  constructor(x?: number, y?: number) {
+    this.x = x ?? 0;
+    this.y = y ?? 0;
+  }
+
+  toString(): string {
+    return `(${this.x}, ${this.y})`;
+  }
+
+  distance(): number;
+  distance(other: Point): number;
+  distance(x: number, y: number): number;
+
+  distance(other?: Point | number, y?: number): number {
+    if (typeof other === 'number' && y !== undefined) {
+      return this.calculateDistanceTo(other, y);
+    }
+    if (other === undefined && y === undefined) {
+      return this.calculateDistanceTo(0, 0);
+    } else {
+      const point = other as Point;
```

Comment:
This local variable seems a bit redundant, as for me)

Reply:
line 27: "other?: Point | number". test fails without explicitly indicating that in this case other is Point.